### PR TITLE
Editor can publish articles

### PR DESCRIPTION
**Pull request by arunbhalli**
**PT-Story:**  https://www.pivotaltracker.com/story/show/178264787
### User Story 
``` 
As an Editor,
In order to make sure that the articles have good standards
I would like only editors to be able to publish articles. 
``` 
## Changes proposed in this pull request: 
* Add Editor state depending on role attribute from authentication response
* Add published attribute to fixture
* Make a published column in the table
* Make the edit button into an 'action' button that has to publish, edit
* `PUT` request to route `api/articles/:id` for the publish

## What I have learned working on this feature: 
* Popup component
* Window.store dispatching in Cypress

## Screenshots (Client) 
image

### CODE REVIEW

**Review by LucasKnudsen — APPROVED** [review]

File: src/index.js [hunk 1]
```diff
@@ -21,3 +21,7 @@ ReactDOM.render(
 );
 
 reportWebVitals();
+
+if (window.Cypress) {
+  window.store = store;
+}
```
Comment (1CIM) [inline src/index.js:27]: This nice dude again :3

**Review by 1CIM — APPROVED** [review]
Good to get that prettier thing set now 😅

File: src/components/ArticleTable.jsx [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+              </Table.Cell>
+              <Table.Cell>
+                <Rating
+                  data-cy='rating'
+                  icon='star'
+                  size='tiny'
+                  defaultRating={
+                    article.rating
+                      ? article.rating
+                      : Math.floor(Math.random() * 6)
```
Comment (DavDan1) [inline src/components/ArticleTable.jsx]: Why did you give math.random?

Comment (offcast-dev) [inline src/components/ArticleTable.jsx]: What is this? Does it override the articles rating from API? If it is done just to display different ratings we better do it in the seed.

File: src/modules/Articles.js [hunk 3]
```diff
@@ -27,7 +27,7 @@ const Articles = {
   async show(id) {
     try {
       let response = await axios.get(`/articles/${id}`, {
-        headers: getFromLocalStorage(),
+        headers: { ...getFromLocalStorage(), source: 'admin-system' },
```
Comment (DavDan1) [inline src/modules/Articles.js:30]: how does these dots  work?

Comment (offcast-dev) [inline src/modules/Articles.js:30]: can you explain 3 dots?

**Review by DavDan1 — APPROVED** [review]

**Review by offcast-dev — CHANGES_REQUESTED** [review]
I would remove random rating

File: src/components/ArticleTable.jsx (hunk 2 above)
Comment (lukeperrin10) [reply src/components/ArticleTable.jsx]: This needs to be removed, it was used to simulate a rating before the rating was finished on the backend. Miss between switch of teams I believe.

File: src/modules/Articles.js (hunk 3 above)
Comment (lukeperrin10) [reply src/modules/Articles.js:30]: the 3 dots are destructing the `getFromLocalStorage` and then injecting the `source: admin-system` into the headers before sending them to the back end.

**Review by offcast-dev — APPROVED** [review]

File: cypress/integration/editorCanPublishArticles.feature.js [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+        .its('store')
+        .invoke('dispatch', {
+          type: 'LOG_IN',
+          payload: { fullName: 'Mr. Editor', role: 'editor' },
+        });
+    });
+
+    it('is expected to show action button in dashboard', () => {
+      cy.get('[data-cy=action-btn]').first().click();
+      cy.get('#publish-btn').click();
```
Comment (tochman) [inline cypress/integration/editorCanPublishArticles.feature.js]:
```
cy.get('button').contains('Publish').click()
```

**Pull request merged**